### PR TITLE
More convinent one time parameter access

### PR DESCRIPTION
**Pull request by IamTheCarl**
TLDR: If you just want to read a single ROS parameter and not receive updates for it in the future, it's now this easy:
```rust
let serial_device: String = node
    .get_parameter("serial_device")
    .context("Failed to get serial device name.")?; // Graceful error handling with the anyhow crate.

let baud_rate: Option<i64> = node
    .get_parameter("baud_rate")
    .context("Failed to get device baud rate.")?;
let baud_rate = baud_rate.unwrap_or(115200); // By fetching an optional type, we can assume a default when the value is not set.
```

I write a lot of ROS drivers. The usual way I configure the drivers is to load their settings from ROS params.
The current method to access a ROS parameter once is quite tedious and I wanted to make it easy.

The following is an explanation of my development process and why each commit was made.

The current way to read a parameter just once on startup looks something like this:
```rust
let params = node.params.lock().unwrap();
let serial_device = params.get("serial_device").unwrap();
let serial_device = match serial_device.value.clone() {
    ParameterValue::String(value) => value,
    _ => panic!("I don't feel like dealing with this error."),
};
```

That's usable but ugly, so I set out to try and make it less ugly.
My first step was to generalize converting the type of a value.
That resulted in this:
```rust
let parameters = node.params.lock().unwrap();
let serial_device: String = parameters
    .get("serial_device")
    .context("Serial device name was not provided.")?
    .value
    .clone()
    .try_into()
    .context("`serial_device` parameter is of wrong type")?;
```

Next was to add a "get_parameter" method to the node, simplifying to this:
```rust
let serial_device: String = node
    .get_parameter("serial_device")
    .context("Failed to get serial device name.")?;
```

Now that looks pretty good, except for when you want to give a parameter a default value.
```rust
let baud_rate: i64 = match node.get_parameter("baud_rate") {
    Ok(baud_rate) => Ok(baud_rate),
    Err(r2r::Error::ParameterNotSet { .. }) => Ok(115200), // Default value when not set.
    Err(error) => Err(error),
}
.context("Failed to get device baud rate.")?;
```

To fix that issue, I decided to not make `ParameterNotSet` its own error and instead make it a type error. "I was expecting an Integer but got a NotSet" instead of "Expected parameter is not set". By doing this, I could then have a `try_into` for Option<T> types, enabling the following patern:
```rust
let baud_rate: Option<i64> = node
    .get_parameter("baud_rate")
    .context("Failed to get device baud rate.")?;
let baud_rate = baud_rate.unwrap_or(115200);
```

### CODE REVIEW

**Comment by m-dahl** [comment]
Thanks for you contribution! This is a nice improvement in usability and I am happy to merge it.

Before that, can I ask you to add one or two examples based on your writeup to examples/paramters.rs?

**Comment by IamTheCarl** [comment]
Hopefully that explains it well. I tried to make the comments match the styling of the ones already there.

**Comment by m-dahl** [comment]
Thanks!
